cdap-app-fabric/src/main/java/co/cask/cdap/internal/profile/ProfileService.java

```diff
@@ -155,23 +167,74 @@ public void createIfNotExists(ProfileId profileId, Profile profile) {
   }
 
   /**
-   * Deletes the profile from the profile store
+   * Deletes the profile from the profile store. Profile deletion must satisfy the following:
+   * 1. Profile must exist and must be DISABLED
+   * 2. There must entities assigned to this profile, now the assignment comes from Preference Store and Schedule store
```

Comment:
Profile must not be assigned to any entities. Profiles can be assigned to an entity by setting a preference or a schedule property.